layers_from_onnx/tune_eval_vm.py

```diff
@@ -0,0 +1,673 @@
+        ((1, 1024, 50, 50), (2048, 1024, 1, 1), (0, 0, 0, 0), (2, 2), 0),
+        ((1, 2048, 25, 25), (512, 2048, 1, 1), (0, 0, 0, 0), (1, 1), 1),
+        ((1, 2048, 25, 25), (256, 2048, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 1024, 50, 50), (256, 1024, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 512, 100, 100), (256, 512, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 200, 200), (256, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 100, 100), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 0),
+        ((1, 256, 200, 200), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 0),
+        ((1, 256, 25, 25), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 0),
+        ((1, 256, 13, 13), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 0),
+        ((1, 256, 25, 25), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 1),
+        ((1, 256, 50, 50), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 0),
+        ((1, 256, 50, 50), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 1),
+        ((1, 256, 100, 100), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 1),
+        ((1, 256, 200, 200), (256, 256, 3, 3), (1, 1, 1, 1), (1, 1), 1),
+        ((1, 256, 13, 13), (12, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 25, 25), (12, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 50, 50), (12, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 100, 100), (12, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+        ((1, 256, 200, 200), (12, 256, 1, 1), (0, 0, 0, 0), (1, 1), 0),
+    ]
+    nms = [
+        ((1, 1000, 4), (1, 1, 1000), 2000, 0.7, 0),
+        ((1, 507, 4), (1, 1, 507), 2000, 0.7, 0)
+    ]
+    return batch_norm, bias_add, nms
+    
+def generate_model_bn(dtype, input_shape, filter_shape, padding, strides, relu, leaky=False):
+    dtype = "float32" if args.dtype == "float32" else "float16"
+    shape_dict = {
+        "input": input_shape,
+        "weight": filter_shape,
+        "bn_gamma0": (filter_shape[0],),
+        "bn_beta0": (filter_shape[0],),
+        "bn_mean0": (filter_shape[0],),
+        "bn_var0": (filter_shape[0],),
+    }
+    input = tvm.relay.var("input", shape=input_shape, dtype=dtype)
+    weight = tvm.relay.var("weight", shape=filter_shape, dtype=dtype)
+    bn_gamma0 = tvm.relay.var("bn_gamma0", relay.TensorType((filter_shape[0],), dtype))
+    bn_beta0 = tvm.relay.var("bn_beta0", relay.TensorType((filter_shape[0],), dtype))
+    bn_mmean0 = tvm.relay.var("bn_mean0", relay.TensorType((filter_shape[0],), dtype))
+    bn_mvar0 = tvm.relay.var("bn_var0", relay.TensorType((filter_shape[0],), dtype))
+
+    channels = filter_shape[0]
+    kernel_size = (filter_shape[2], filter_shape[3],)
+    
+    
+    D = relay.nn.conv2d(input, weight, padding=padding, strides=strides, channels=channels, kernel_size=kernel_size)
+    D = relay.op.nn.batch_norm(D, bn_gamma0, bn_beta0, bn_mmean0, bn_mvar0)
+    D2 = D[0]
+    if relu:
+        D2 = relay.op.nn.relu(data=D2)
+        if leaky:
+            D2 = relay.op.nn.leaky_relu(data=D2)
```

Comment:
Just to double-check. Is everything correct with indents here?

Reply:
Yes, it could be two-leveled if statement or just two sequential if statements, but with "leaky relu" check at the end of them, so it overwrites normal relu if "leaky=True".